### PR TITLE
Add `eslint` option to `new-rule` script in eslint-bridge

### LINKED ISSUE #3176: Update `eslint-bridge` `new-rule` script: add option for ESLint-imported rules
Instead of undoing some changes after running the script, add a `eslint` option when calling it.

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**
Fixes #3176

### CODE REVIEW

File: docs/DEV.md [hunk 1]
```diff
@@ -65,12 +65,12 @@ Note that you can fix the port in `orchestrator.properties files`, e.g. `orchest
 java -jar <location of rule-api jar> generate -rule S1234 [-branch <RSPEC branch>]
 ```
 
-2. Generate other files required for a new rule (use existing <ESLint-style rulekey> if the rule is already covered by ESLint or its plugins)
+1. Generate other files required for a new rule. If the rule is already covered by ESLint or its plugins, use the existing <ESLint-style rulekey> and add the `eslint` option.
```
Comment (vilchik-elena) [inline docs/DEV.md:68]: why `1.`?

File: eslint-bridge/tools/newRule.ts [hunk 2]
```diff
@@ -301,8 +304,30 @@ function replace(text: string, dictionary: { [x: string]: string }): string {
...
+ * @param dict 
+ */
 function copyWithReplace(src: string, dest: string, dict: { [x: string]: string }) {
   const content = fs.readFileSync(src, 'utf8');
   const newContent = replace(content, dict);
   fs.writeFileSync(dest, newContent);
 }
+
+function parseOptions(options: string[]) {
+  const [ ESLINT_OPTION ] = ['eslint' ]
```
Comment (vilchik-elena) [inline eslint-bridge/tools/newRule.ts:321]: what is this destructuring hell?

File: eslint-bridge/tools/newRule.ts [hunk 3]
```diff
@@ -35,12 +35,13 @@ run();
...
 function run() {
-  if (process.argv.length !== 4) {
-    throw new Error(`Invalid number of arguments: expected 2, but got ${process.argv.length - 2}`);
+  if (process.argv.length < 4) {
+    throw new Error(`Insufficient number of arguments: expected at least 2, but got ${process.argv.length - 2}`);
   }
 
   const rspecId = process.argv[2];
   const ruleNameDash = process.argv[3];
+  const { isEslint } = parseOptions(process.argv.slice(4));
```
Comment (vilchik-elena) [inline eslint-bridge/tools/newRule.ts:44]: why can't we simply do `const isEslint = process.argv[4] === 'eslint'`?

**Review by vilchik-elena — CHANGES_REQUESTED** [review]

File: docs/DEV.md (hunk 1 above)
Comment (ilia-kebets-sonarsource) [reply docs/DEV.md:68]: IDE changed it for me. I didn't see it.

File: eslint-bridge/tools/newRule.ts (hunk 2 above)
Comment (ilia-kebets-sonarsource) [reply eslint-bridge/tools/newRule.ts:321]: I implemented it first with 2 options, hence the over-engineering. I removed `-quickfix` afterwards, without cleaning up the rest.

File: eslint-bridge/tools/newRule.ts [hunk 4]
```diff
@@ -61,7 +62,10 @@ function run() {
   updateCheckList();
 
   /** Creates rule typescript source and test files from templates */
-  function createTsFiles() {
+  function createTsFiles(isEslint: boolean) {
+    
+    if ( isEslint ) { return; }
```
[the remark this answers is not in the document]
Comment (ilia-kebets-sonarsource) [reply eslint-bridge/tools/newRule.ts:67]: - the newlines chaos is due to retrieving it from the commits of the other PR, I didn't check whitespaces.
- I like the pattern of exiting functions quickly on some cases like `if (invalid) return;`, Sonar made me put these brackets. I guess I could adopt spreading it on 3 lines, but I'm not a fan.
- the semicolon was lost in some refactoring. As this file is not flagged correctly in VScode, I didn't pay attention to it.

**Review by vilchik-elena — APPROVED** [review]

**Pull request merged**